app/routes.py

```diff
@@ -1,2 +1,196 @@
+
+@videos_bp.route("", methods = ["POST"])
+def videos():
+    request_body = request.get_json()
+    if "title" not in request_body or "release_date" not in request_body or "total_inventory" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)    
+    else: 
+        new_video = Video(title=request_body["title"], 
+                        release_date=request_body["release_date"],
+                        total_inventory=request_body["total_inventory"],
+                        available_inventory=request_body["total_inventory"]) 
+        db.session.add(new_video)
+        db.session.commit() 
+        return make_response({"id": new_video.id}, 201)
+
+@videos_bp.route("/<id>", methods = ["GET"])
+def get_one_video(id):
+    video = Video.query.get_or_404(id)
+    return  make_response(video.to_json(), 200)
+
+@videos_bp.route("/<id>", methods = ["PUT"])
+def update_video(id):
+    video = Video.query.get_or_404(id)
+    request_body = request.get_json() 
+    if "title" not in request_body or "release_date" not in request_body or "total_inventory" not in request_body:
+        return make_response({"details": "Invalid data"}, 400)    
+    else:
+        video.title = request_body["title"]
+        video.release_date = request_body["release_date"]
+        video.total_inventory = request_body["total_inventory"]
+        db.session.commit()
+    return make_response(video.to_json(), 200) 
+
+@videos_bp.route("/<id>", methods = ["DELETE"])
+def delete_video(id):
+    video = Video.query.get_or_404(id)
+    
+    db.session.delete(video)
+    db.session.commit()
+
+    return make_response({"id": video.id}, 200)
+
+
+@videos_bp.route("/<id>/rentals", methods = ["GET"])
+def video_rentals(id): 
+    video = Video.query.get_or_404(id)
+    video_rentals = Rental.query.filter_by(video_id = video.id)
+    customers_by_video = []
+    for rental in video_rentals:
+        customer=Customer.query.get_or_404(rental.customer_id)
+        
+        customers_by_video.append({
+                        "due_date": rental.due_date,
+                        "name": customer.name,
+                        "phone": customer.phone,
+                        "postal_code": str(customer.postal_code)})
+    return jsonify(customers_by_video), 200
+
+
+# RENTAL
+
+@rentals_bp.route("/check-out", methods = ["POST"])
+def check_out():
+    request_body = request.get_json()
+    customer_id = request_body.get("customer_id")
+    video_id=request_body.get("video_id")
+
+    customer = Customer.query.get_or_404(customer_id)
+    video = Video.query.get_or_404(video_id)
+
+    if video.available_inventory == 0: return make_response({"errors": "No video available"}, 400)
+
+    customer.videos_checked_out_count += 1
+    video.available_inventory -= 1
+    
+    rental = Rental(customer_id = customer.id, video_id=video.id, due_date = datetime.now() + timedelta(days=7))
+    db.session.add(rental)
+    db.session.commit() 
```

Comment:
Great job in setting up the check-out route! Optionally, you can keep the `routes.py` page to process request body data and then call a helper method in `Rental`. The helper method can have the logic to make the new rental.